### ISSUE TITLE
Define new objective function

### ISSUE THREAD

**Issue by zyzhong** [dmlc/xgboost#1212]
It seems that xgboost takes the Taylor expansion of the loss function up to the second order. So can we define our own loss function? How to modify the code to achieve that? BTW, I use python interface of xgboost

**Comment by zyzhong** [comment]
got help from [here](https://github.com/dmlc/xgboost/blob/master/demo/guide-python/custom_objective.py)
